apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/KotlinCompileCompat.kt

```diff
@@ -0,0 +1,14 @@
+package com.apollographql.apollo.gradle.internal
+
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+// The signature of this function changed in Kotlin 1.7, so we invoke it reflectively
+// to support both.
+// 1.6.x: `fun source(vararg sources: Any): SourceTask
+// 1.7.x: `fun source(vararg sources: Any)
```

Comment:
Attribute the fix in the source too?

```suggestion
// From https://github.com/square/wire/pull/2273
// The signature of this function changed in Kotlin 1.7, so we invoke it reflectively
// to support both.
// 1.6.x: `fun source(vararg sources: Any): SourceTask
// 1.7.x: `fun source(vararg sources: Any)
```